apps/etterlatte-beregning/src/main/kotlin/beregning/BeregningRepository.kt

```diff
@@ -74,12 +74,16 @@ class BeregningRepository(private val dataSource: DataSource) {
                         "sakId" to overstyrBeregning.sakId,
                         "beskrivelse" to overstyrBeregning.beskrivelse,
                         "tidspunkt" to overstyrBeregning.tidspunkt.toTimestamp(),
+                        "status" to overstyrBeregning.status.name,
                     ),
             ).let { query ->
                 tx.run(query.asUpdate)
             }
         }
 
+        if (overstyrBeregning.status == OverstyrBeregningStatus.UGYLDIG) {
```

Comment:
Skjønner ikke helt hvorfor denne kommer etter man oppretter overstyringen?

Reply:
Tanken er å opne for å opprette ugyldige overstyringar, viss vi av ein eller annan grunn ønskjer det.

Reply:
Jo men er den i riktig metode? Skal den ikke være i hentOverstyrBeregning ikke opprettOverstyrBeregning?

Reply:
Ah, sånn ja. I hent har vi eit filter i sql-spørringa, som gjer at vi berre hentar ut dei som er gyldige.

Her opprettar vi først, og så kallar vi hent eit par linjer under denne. Det er da det kræsjar for dei nyoppretta ugyldige, kor hent returnerer null (som forventa), og da slår checkNotNull-en til.

Reply:
Åja ok. Klarte ikke å se hvor det filtreres i hentinga.

Men hva skjer nå hvis saksbehandler har overstyring me ugyldig og ønsker å overstyre på nytt i en senere behandling? Da vil det insertes enda en overstyring da? En gyldig og en ugyldig?

Reply:
Ai, godt poeng. Da vil det nok tryne på duplicate key, for sak_id er primærnøkkel i tabellen. Trengs kanskje ein `ON CONFLICT` i inserten for å omgå det.

(Eventuelt skrive om tabellen til å ikkje bruke sak_id som primærnøkkel, men det trur eg er meir arbeid enn det er verdt)

Reply:
Vil man ikke uansett hente hele objektet, uavhengig om det er overstyrt eller ikke? Så er det opp til service hva man ønsker å gjøre videre basert på status?

Reply:
Jo, kanskje. Men eg har eigentleg ikkje lyst til å spre dette konseptet lengre ut i kodebasen enn nødvendig, iallfall i denne omgang. For meg er ugyldig her mest ein parallell til feilført eller avbrutt, altså ikkje noko som vi ønskjer å forholde oss til i "vanleg" kode.

Reply:
Opprinneligvis var det å ha det PK på sak_id helt bestemt - men det er fra den tiden kun utviklere i basen kunne sette noe til overstyrt. Kan hende det er verdt en nyvurdering på - vi diskuterte om det burde være på behandling for å kunne endre fra overstyrt til ikke overstyrt bl.a. på det tidspunktet.